### PR TITLE
EZP-21008: Replaced assertType() with assertInternalType()

### PR DESCRIPTION
**Pull request by jeromegamez**
The `assertType()` method has been removed in favor of `assertInternalType()`. `assertType()` has been ["silently" deprecated](https://github.com/sebastianbergmann/phpunit/issues/77#issuecomment-619328) in PHPUnit 3.5 and removed in 3.6.

So, the Unit Tests for eZ Comments will continue to work in PHPUnit 3.5 (the minimum version for eZ Publish), but later will fail (which they did when I installed a new eZ Publish Legacy from the git master branch with the demo extension).

This patch changes `assertType()` calls to `assertInternalType()`.

Cheers
:octocat: Jérôme

https://jira.ez.no/browse/EZP-21008

### CODE REVIEW

**Comment by andrerom** [comment]
+1

**Comment by lolautruche** [comment]
Why not using `assertInstanceOf()` ?

**Comment by andrerom** [comment]
@jeromegamez As mentioned by @lolautruche, your second commit should preferably be amended to use assertInstanceOf() instead of assertTrue().

**Comment by jeromegamez** [comment]
@lolautruche Oh, right, I forgot about this one! I'll change it!

@andrerom How do you amend to an already pushed commit? The only way I know is to rollback and force push, but I also know this is considered bad practice.

**Comment by andrerom** [comment]
@jeromegamez It's a very bad practice when working togheter with others, but as we also value clean version history we often do it on PR branches. It's a `git add -u`, `git commit --amend` and a `git push --force`. If you at some point need to edit a commit which is not the last on a PR branch, then `git rebase -i master` is your friend.

**Comment by jeromegamez** [comment]
It worked! :). Thank you for the hints!

**Comment by andrerom** [comment]
Congrats, just remember: With great power comes great responsibility :P
@lolautruche ?

File: packages/ezcomments_extension/ezextension/ezcomments/tests/ezcomnotification_test.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function testCreateObject()
     public function testFetchObject()
     {
         $notification = ezcomNotification::fetch( 1 );
-        $this->assertType( 'ezcomNotification', $notification );
+        $this->assertTrue( $notification instanceof ezcomNotification );
```
Comment (lolautruche) [inline packages/ezcomments_extension/ezextension/ezcomments/tests/ezcomnotification_test.php]: Still missing `assertInstanceOf()`

**Comment by jeromegamez** [comment]
Sorry @lolautruche , I missed the last one, thanks for pointing it out. Now all should be set.

**Comment by lolautruche** [comment]
+1

**Comment by andrerom** [comment]
So we are set, the unit tests runs with this as well? :)

**Comment by jeromegamez** [comment]
To be honest, the tests mostly fail because of the sharedFixture not being set, but this patch fixes the issue that the tests quit with a PHP error about the undefined assertType() method. 

So I think the issue of this PR is solved.

**Comment by andrerom** [comment]
Thanks @jeromegamez :)

**Pull request merged**